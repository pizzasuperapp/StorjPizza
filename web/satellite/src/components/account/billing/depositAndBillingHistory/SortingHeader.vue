// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="sort-header-container">
        <div class="sort-header-container__item date">
            <p class="sort-header-container__item__name">DATE</p>
        </div>
        <div class="sort-header-container__item description">
            <p class="sort-header-container__item__name">DESCRIPTION</p>
        </div>
        <div class="sort-header-container__item status">
            <p class="sort-header-container__item__name">STATUS</p>
        </div>
        <div class="sort-header-container__item amount">
            <p class="sort-header-container__item__name">AMOUNT</p>
        </div>
        <div class="sort-header-container__item download">
            <p class="sort-header-container__item__name">DOWNLOAD</p>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

// @vue/component
@Component
export default class SortingHeader extends Vue {}
</script>

<style scoped lang="scss">
    .sort-header-container {
        display: flex;
        width: 100%;
        padding-bottom: 32px;

        &__item {
            text-align: left;
            min-width: 20%;

            &__name {
                font-family: 'font_medium', sans-serif;
                font-size: 14px;
                line-height: 19px;
                color: #adadad;
                margin: 0;
            }
        }
    }

    .description {
        min-width: 31%;
    }

    .status {
        min-width: 17%;
    }

    .amount {
        min-width: 22%;
        margin: 0;
    }

    .download {
        text-align: right;
        margin: 0;
        min-width: 10%;
    }
</style>
