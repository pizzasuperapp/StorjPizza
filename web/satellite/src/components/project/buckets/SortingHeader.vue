// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="sort-header-container">
        <div class="sort-header-container__item name">
            <p class="sort-header-container__item__name">BUCKET NAME</p>
        </div>
        <div class="sort-header-container__item">
            <p class="sort-header-container__item__name">STORAGE</p>
        </div>
        <div class="sort-header-container__item">
            <p class="sort-header-container__item__name">BANDWIDTH</p>
        </div>
        <div class="sort-header-container__item">
            <p class="sort-header-container__item__name">OBJECTS</p>
        </div>
        <div class="sort-header-container__item">
            <p class="sort-header-container__item__name">SEGMENTS</p>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

// @vue/component
@Component
export default class SortBucketsHeader extends Vue {}
</script>

<style scoped lang="scss">
    .sort-header-container {
        display: flex;
        background-color: #fff;
        padding: 35px 40px;
        align-items: center;
        border-bottom: 1px solid #c7cdd2;
        border-radius: 6px 6px 0 0;

        &__item {
            width: 20%;
            margin: 0;
            text-align: left;

            &__name {
                margin: 0;
                font-family: 'font_medium', sans-serif;
                font-size: 14px;
                line-height: 20px;
                color: #acacac;
                letter-spacing: 0.03em;
            }
        }
    }

    .name {
        width: 40%;
    }
</style>
