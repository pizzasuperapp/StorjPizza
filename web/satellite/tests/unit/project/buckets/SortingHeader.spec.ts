// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import SortingHeader from '@/components/project/buckets/SortingHeader.vue';

import { createLocalVue, shallowMount } from '@vue/test-utils';

const localVue = createLocalVue();

describe('SortingHeader.vue', () => {
    it('renders correctly', (): void => {
        const wrapper = shallowMount(SortingHeader, {
            localVue,
        });

        expect(wrapper).toMatchSnapshot();
    });
});
