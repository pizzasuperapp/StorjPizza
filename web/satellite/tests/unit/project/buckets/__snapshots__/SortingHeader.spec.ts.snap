// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingHeader.vue renders correctly 1`] = `
<div class="sort-header-container">
  <div class="sort-header-container__item name">
    <p class="sort-header-container__item__name">BUCKET NAME</p>
  </div>
  <div class="sort-header-container__item">
    <p class="sort-header-container__item__name">STORAGE</p>
  </div>
  <div class="sort-header-container__item">
    <p class="sort-header-container__item__name">BANDWIDTH</p>
  </div>
  <div class="sort-header-container__item">
    <p class="sort-header-container__item__name">OBJECTS</p>
  </div>
  <div class="sort-header-container__item">
    <p class="sort-header-container__item__name">SEGMENTS</p>
  </div>
</div>
`;
